pumps/prometheus.go

```diff
@@ -197,6 +185,27 @@ func (p *PrometheusPump) Init(conf interface{}) error {
 	return nil
 }
 
+//InitCustomMetrics initialise custom prometheus metrics based on p.conf.CustomMetrics and add them into p.allMetrics
+func (p *PrometheusPump) InitCustomMetrics() error {
+	if len(p.conf.CustomMetrics) > 0 {
+		customMetrics := []*PrometheusMetric{}
+		for i := range p.conf.CustomMetrics {
+			newMetric := &p.conf.CustomMetrics[i]
+			newMetric.aggregatedObservations = p.conf.AggregateObservations
+			errInit := newMetric.InitVec()
+			if errInit != nil {
+				return errInit
```

Comment:
@tbuchaillot if we have an error is ok that we just return? and do not continue in the next iteration?

Reply:
@sredxny do you suggest that we just log the name of the invalid one and add the others?

Reply:
@sredxny changed to work in a non-blocking when missconfigured. good suggestion